LeaderboardBackend/Models/Entities/NumericalMetric.cs

```diff
@@ -0,0 +1,96 @@
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+
+namespace LeaderboardBackend.Models.Entities;
+
+/// <summary>
+///     Represents a `NumericalMetric` for a `Category`.
+/// </summary>
+public class NumericalMetric : BaseEntity
+{
+	/// <summary>
+	///     The unique identifier of the `NumericalMetric`.<br/>
+	///     Generated on creation.
+	/// </summary>
+	public long Id { get; set; }
+
+	/// <summary>
+	///     The display name of the `NumericalMetric`.
+	/// </summary>
+	/// <example>Foo</example>
+	[NotNull]
+	[Required]
+	public string? Name { get; set; }
+
+	/// <summary>
+	///     The minimum value of the `NumericalMetric`. The default is 0.
+	/// </summary>
+	[Required]
+	public long Min { get; set; }
+
+	/// <summary>
+	///     The maximum value of the `NumericalMetric`. The default is `Min` + 1.
+	/// </summary>
+	[Required]
+	public long Max { get; set; }
+
+	/// <summary>
+	///     A collection of `Category`s scoped to the `NumericalMetric`.
+	/// </summary>
+	[MinLength(1)]
+	public List<Category>? Categories { get; set; }
+
+	/// <summary>
+	///     A collection of `Run`s scoped to the `NumericalMetric`.
+	/// </summary>
+	public List<Run>? Runs { get; set; }
```

Comment:
This doesn't seem right. How can a metric have runs that belong to it? Is this just the runs that are using this metric?

Reply:
The latter. Once again looking at the LucidChart, these - IntegerMetrics I guess? - are a many-to-many with runs. EF Core conventions call for both entities to add an `ICollection` of the other. You should've already seen a `public List<NumericalMetric>? NumericalMetrics { get; set; }` line in `Run.cs`.